sphinx/ext/intersphinx.py

```diff
@@ -466,6 +470,145 @@ def missing_reference(app: Sphinx, env: BuildEnvironment, node: pending_xref,
     return resolve_reference_detect_inventory(env, node, contnode)
 
 
+class IntersphinxDispatcher(CustomReSTDispatcher):
+    """Custom dispatcher for external role.
+
+    This enables :external:***:/:external+***: roles on parsing reST document.
+    """
+
+    def role(self, role_name: str, language_module: ModuleType, lineno: int, reporter: Reporter
+             ) -> Tuple[RoleFunction, List[system_message]]:
+        if len(role_name) > 9 and role_name.startswith('external') and role_name[8] in ':+':
```

Comment:
How abou this?
```suggestion
        if len(role_name) > 9 and role_name.startswith(('external:', 'external+')):
```

Reply:
Good idea, didn't know you could give a tuple to ``startswith``.